database/migrations/default/1736197283308_jurisdiction_refactor/up.sql

```diff
@@ -0,0 +1,143 @@
+alter table geo.atd_jurisdictions add column jurisdictions_id integer;
+
+create or replace function public.crashes_set_spatial_attributes()
+returns trigger
+language plpgsql
+as $function$
+begin
+    if (new.latitude is not null and new.longitude is not null) then
+        -- save lat/lon into geometry col
+        new.position = st_setsrid(st_makepoint(new.longitude, new.latitude), 4326);
+        --
+        -- get location polygon id
+        --
+        if (new.rpt_road_part_id != 2 and upper(ltrim(new.rpt_hwy_num)) in ('35', '183','183A','1','290','71','360','620','45','130')) then
+            -- use level 5 polygon
+            new.location_id = (
+                select
+                    location_id
+                from
+                    public.atd_txdot_locations
+                where
+                    location_group = 2 -- level 5
+                    and st_contains(geometry, new.position)
+                limit 1);
+        else
+            -- use the other polygons
+            new.location_id = (
+                select
+                    location_id
+                from
+                    public.atd_txdot_locations
+                where
+                    location_group = 1 -- not level 5
+                    and st_contains(geometry, new.position)
+                limit 1);
+        end if;
+
+        raise debug 'found location: % compared to previous location: %', new.location_id, old.location_id;
+        --
+        -- check if in austin full purpose jurisdiction
+        --
+        new.in_austin_full_purpose = EXISTS (
+            select 1 
+            from geo.atd_jurisdictions 
+            where JURISDICTION_LABEL = 'AUSTIN FULL PURPOSE' AND  ST_Intersects(geometry, new.position)
```

Comment:
I think you can get some meaningful speed-up on this where clause by doing:

```
where 
JURISDICTION_LABEL = 'AUSTIN FULL PURPOSE' 
AND geometry && new.position
AND  ST_Intersects(geometry, new.position)
```

`&&` is the "has intersecting bounding boxes" operator, and it's a zillion times faster to see if a point and a box overlap than a point and an arbitrary polygon. By doing this blazing fast operation first, you can skip over checking the `st_intersect()` on all the pairs of geometries that don't have overlapping bounding boxes, knowing that they couldn't intersect no matter what.

Also PG's GIST indexing on geometry is aimed at exactly this -- it stores the bounding boxes of geometries as its index data, so `&&` can get really lightning fast.

Reply:
i ran those two statements on staging and the results are consistent: removing the `&&` check results in better performance by nearly 50% 😵

Reply:
John - What a twist! Running those two queries on my machine come up with a less dramatic difference in timing 765 ms vs 515 ms, but the extra clause I'm suggesting really does seem to be the difference that is slowing it down.  I'm fully confused about why it's doing it this way, but the numbers don't like -- thank you for testing this out and providing such a clear cut set of data. 🙏

Reply:
really truly a twist! i'd love to spend some more time understanding what's going on here. i'd love to re-test this scenario against a table with more rows, like `atd_txdot_locations`....